### PR TITLE
chore: add chart

### PR DESCRIPTION
**Pull request by AnguIar**

### CODE REVIEW

File: nginx/docker-image/nginx-config/status_site.conf [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+        server_name  localhost;
+
+        location /nginx_status { 
+                opentelemetry off;
+                access_log off;
+                stub_status;
+                #allow 127.0.0.1; #only allow requests from localhost
+                #deny all; #deny all other hosts 
+        }
+}
```
Comment (shimoncohen) [inline nginx/docker-image/nginx-config/status_site.conf]: Add a line at the end of the file.

File: nginx/docker-image/subset.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+cat /otel/otel-nginx-template.toml | envsubst > /otel/otel-nginx.toml
```
Comment (shimoncohen) [inline nginx/docker-image/subset.sh]: Add a line at the end of the file.

File: nginx/helm/templates/route.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{{- $releaseName := .Release.Name -}}
+{{- $chartName := include "nginx.name" . -}}
+{{- $cloudProviderFlavor := include "nginx.cloudProviderFlavor" . -}}
+{{- $serviceName := include "service.fullname" . -}}
+{{- if and (and (.Values.enabled) (eq $cloudProviderFlavor "openshift")) (.Values.route.enabled) -}}
```
Comment (shimoncohen) [inline nginx/helm/templates/route.yaml:3]: Please move to begining of file.

File: nginx/helm/config/log_format.conf [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    ~(.*)\.(.*) $1$2000000;
+}
+
+log_format main escape=json 
+'{'
+    '"Timestamp":$nanosec,'
+    '"Attributes":{'
+        '"mapcolonies.time_local":"$time_local",'
+        {{ if .Values.authentication.enabled }}
+        '"mapcolonies.http.auth.token.client_name":"$jwt_payload_sub",'
```
Comment (shimoncohen) [inline nginx/helm/config/log_format.conf:11]: Please come back with explenation about this and check about the differences between `Attributes` and `Resource`.

File: nginx/docker-image/README.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# nginx working image for openshit
+This image is a version of nginx that works in openshit environment.
+All the permissions were changed so a user under the group `root` can run the container.
```
Comment (shimoncohen) [inline nginx/docker-image/README.md]: Please update README.

**Review by shimoncohen — CHANGES_REQUESTED** [review]

File: nginx/helm/templates/route.yaml (hunk 3 above)
Comment (AnguIar) [reply nginx/helm/templates/route.yaml:3]: In line 5 there's a condition that checks value of `$cloudProviderFlavor` which is defined couple of lines before, so we need the `if` statement in here.

File: nginx/README.md [hunk 6]
```diff
@@ -1,20 +1,60 @@
...
+### Authroization Mechanism 
+Since we are using [Open Policy Agent](https://www.openpolicyagent.org/) (aka `OPA`) as our gatekeeper, it's necessary to integrate Nginx with it. 
+* The docker image does contains the `auth.js` file, which is responsible for handling requests that require authorization **but** the Nginx server does not actually make the authorization process - we commented the code section that's doing it.
+
+### Log Format
+The docker image provides default log format  (`/etc/nginx/log_format`). It's not possible to extend the log format, so if you'd want to add/remove certain fields you have to override it.
+
+
+## Helm Chart
+There is also an Helm Chart for deploying this Nginx in an Openshift environment (let alone any K8S environment).
```
Comment (shimoncohen) [inline nginx/README.md]: Additionaly there is a Helm Chart...

File: nginx/README.md [hunk 7]
```diff
@@ -1,20 +1,60 @@
...
+
+# Nginx Docker Image & Helm Chart for Openshift
+
+This repository consists of two things:
+1. Nginx `Dockerfile` and its necessary assets for building
+2. Nginx `Helm Chart` 
+
+
+## Docker Image
+We are using `nginxinc/nginx-unprivileged` as a base image in order to run Nginx with non-root privileges so it's possible to run in it an Openshift cluster (as we know, Openshift does not allow running containers with root privileges).
```
Comment (shimoncohen) [inline nginx/README.md:10]: ... to run it in an ...

File: nginx/README.md [hunk 8]
```diff
@@ -1,20 +1,60 @@
...
+Besides that the `Dockerfile` is pretty straight forward so you can open it up and view it by yourself.
```
Comment (shimoncohen) [inline nginx/README.md]: ... so you can check it out yourself.

File: nginx/README.md [hunk 9]
```diff
@@ -1,20 +1,60 @@
...
+| ---------------------------- | ---------------------------------------------- | ---- |
+| `OTEL_SERVICE_NAME` | Name of service | `nginx-proxy` |
+| `OTEL_EXPORTER_OTLP_ENDPOINT` | Endpoint address of OpenTelemetry Collector | `localhost:4317` |
+| `OTEL_TRACES_SAMPELR` | Sampling method | `AlwaysOff` |
+| `OTEL_TRACES_SAMPLER_RATIO` | Sampling ratio | `0.1` |
+| `OTEL_TRACES_SAMPLER_PARENT_BASED` | Use parent-based sampling | `false` |
+
+### Authroization Mechanism 
+Since we are using [Open Policy Agent](https://www.openpolicyagent.org/) (aka `OPA`) as our gatekeeper, it's necessary to integrate Nginx with it. 
+* The docker image does contains the `auth.js` file, which is responsible for handling requests that require authorization **but** the Nginx server does not actually make the authorization process - we commented the code section that's doing it.
```
Comment (shimoncohen) [inline nginx/README.md]: The docker image contains ... does not actually handle the ... commented the code section responsible for this logic.

File: nginx/.gitignore [hunk 10]
```diff
@@ -1 +1,2 @@
 *myvalues*
+*.tgz
```
Comment (shimoncohen) [inline nginx/.gitignore:2]: Does this need to be added?

File: nginx/README.md (hunk 7 above)
Comment (shimoncohen) [inline nginx/README.md]: ... to run it in an ...

**Review by shimoncohen — CHANGES_REQUESTED** [review]

File: nginx/helm/templates/deployment.yaml [hunk 11]
```diff
@@ -80,4 +81,25 @@ spec:
...
+          {{- with .Values.prometheusExporter.image }}
+          image: {{ $cloudProviderDockerRegistryUrl }}{{ .repository }}:{{ .tag }}
+          {{- end }}
+          args: ["-nginx.scrape-uri=http://localhost:8081/nginx_status"]
+          imagePullPolicy: {{ .Values.prometheusExporter.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.prometheusExporter.targetPort }}
+              protocol: TCP
+          livenessProbe:
```
Comment (shimoncohen) [inline nginx/helm/templates/deployment.yaml:104]: Please add a warining.

File: nginx/helm/values.yaml [hunk 12]
```diff
@@ -39,6 +39,9 @@ nginx:
       requests:
         cpu: 100m
         memory: 128Mi
+  additionalPodAnnotations:
+    prometheus.io/scrape: 'true'
+    prometheus.io/port: '9113'
```
Comment (shimoncohen) [inline nginx/helm/values.yaml]: Please move to `Helpers`.

**Review by shimoncohen — APPROVED** [review]

File: nginx/helm/templates/route.yaml (hunk 3 above)
Comment (shimoncohen) [reply nginx/helm/templates/route.yaml:3]: 👍🏻

**Pull request merged**